solidity/test/token/TokenBridgeCctp.t.sol

```diff
@@ -312,6 +349,143 @@ contract TokenBridgeCctpTest is Test {
         tbDestination.verify(metadata, message);
     }
 
+    function test_verify_revertsWhen_invalidAmount() public {
+        bytes32 recipient = user.addressToBytes32();
+        Quote[] memory quote = tbOrigin.quoteTransferRemote(
+            destination,
+            recipient,
+            amount
+        );
+
+        vm.startPrank(user);
+        tokenOrigin.approve(address(tbOrigin), quote[1].amount);
+
+        uint64 cctpNonce = tokenMessengerOrigin.nextNonce();
+        tbOrigin.transferRemote{value: quote[0].amount}(
+            destination,
+            recipient,
+            amount
+        );
+        bytes memory message = mailboxDestination.inboundMessages(0);
+
+        // invalid amount := amount + 1
+        uint256 badAmount = amount + 1;
+        bytes memory cctpMessage = _encodeCctpMessage(
+            cctpNonce,
+            cctpOrigin,
+            recipient,
+            badAmount
+        );
+        bytes memory attestation = bytes("");
+        bytes memory metadata = abi.encode(cctpMessage, attestation);
+
+        vm.expectRevert(bytes("Invalid amount"));
+        tbDestination.verify(metadata, message);
+    }
+
+    function test_verify_revertsWhen_invalidRecipient() public {
+        bytes32 recipient = user.addressToBytes32();
+        Quote[] memory quote = tbOrigin.quoteTransferRemote(
+            destination,
+            recipient,
+            amount
+        );
```

Comment:
Nit: Throw some of these configuration steps into some internal helper functions. We can follow up and do this later.

Reply:
done